srtcore/logging.h

```diff
@@ -166,14 +166,20 @@ struct SRT_API LogDispatcher
 
         // See Logger::Logger; we know this has normally 2 characters,
         // except !!FATAL!!, which has 9. Still less than 32.
-        strcpy(prefix, your_pfx);
-
         // If the size of the FA name together with severity exceeds the size,
         // just skip the former.
         if (logger_pfx && strlen(prefix) + strlen(logger_pfx) + 1 < MAX_PREFIX_SIZE)
         {
-            strcat(prefix, ":");
-            strcat(prefix, logger_pfx);
+#if defined(_MSC_VER) && _MSC_VER < 1900
+            _snprintf(prefix, MAX_PREFIX_SIZE, "%s:%s", your_pfx, logger_pfx);
+#else
+            snprintf(prefix, MAX_PREFIX_SIZE + 1, "%s:%s", your_pfx, logger_pfx);
+#endif
+        }
+        else
+        {
+            strncpy(prefix, your_pfx, MAX_PREFIX_SIZE);
+            prefix[MAX_PREFIX_SIZE] = '\0';
```

Comment:
We'd be much better off, if you simply take the length by `strlen` and then copy using `memcpy`. Note that this isn't any performance critical part and it's done once in the beginning of an application. This way you can also use this length to insert `':'` after that. This `strcat` with one colon was actually a little stupid ;)

Reply:
But in that path I thought we don't want the colon? The old code only added the colon when there was room for both your_pfx and logger_pfx which the proposed changes does using the _snprint/snprintf.

Reply:
In that, yes, but you have applied the fix for both paths.

Reply:
Ok, maybe there is some confusion here.

To clarify, the warning I was dealing with was _CRT_SECURE_NO_WARNINGS. This applies to both the original initial strcpy call and then optional 2 x strcat.

The changes are intended to provide the same end result, but using string function calls that don't trigger the above warning when used.

To be honest, there are probably 10 different ways to achieve the same thing, but this seemed as concise as any.